GeoLib/AnalyticalGeometry.h

```diff
@@ -299,6 +299,10 @@ bool parallel(MathLib::Vector3 v, MathLib::Vector3 w);
 bool lineSegmentIntersect (const GeoLib::Point& a, const GeoLib::Point& b,
         const GeoLib::Point& c, const GeoLib::Point& d, GeoLib::Point& s);
 
+std::vector<MathLib::Point3d>
+lineSegmentIntersect2d(MathLib::Point3d const& a, MathLib::Point3d const& b,
```

Comment:
Docu. Especially on return values. Mention also 'ab' is a segment and 'cd' is a segment...